p2p/switch.go

```diff
@@ -273,7 +273,7 @@ func (sw *Switch) OnStop() {
 func (sw *Switch) Broadcast(e Envelope) chan bool {
 	sw.Logger.Debug("Broadcast", "channel", e.ChannelID)
 
-	peers := sw.peers.List()
+	peers := sw.peers.Copy()
```

Comment:
So, here I see a performance cost for this approach. `Broadcast` is used by several reactors (details [here](https://github.com/cometbft/cometbft/blob/main/spec/p2p/reactor-api/p2p-api.md#switch-api)) to broadcast (of course), periodic messages.

I wonder whether we can use the `ForEach` method here too. In particular because we start go routines to actually send messages, so we shouldn't keep the lock for long.